lib/src/repository/traits/pin_code_repository.dart

```diff
@@ -0,0 +1,151 @@
+import 'package:yonomi_platform_sdk/src/queries/pin_code_credential/make_create_pin_code_credential_request/query.req.gql.dart';
+import 'package:yonomi_platform_sdk/src/repository/devices_repository.dart';
+import 'package:yonomi_platform_sdk/src/repository/repository.dart';
+import 'package:yonomi_platform_sdk/src/request/request.dart';
+import 'package:yonomi_platform_sdk/third_party/yonomi_graphql_schema/schema.docs.schema.gql.dart';
+
+typedef PinCodeCredentialInput = GPinCodeCredentialInput;
+typedef PinCodeCredentialListItemInput = GPinCodeCredentialListItemInput;
+typedef _traitNames = GTraitName;
+
+class PinCodeRepository {
+  static PinCodeTrait getPinCodeTrait(trait) {
+    try {
+      return PinCodeTrait(<State>{
+        PinCodeCredentials(
+            trait.state.pinCodeCredentials.reported?.value.edges.toList()),
+      }, <Property>{
+        MaxNumberOfPinCodeCredentials(
+            trait.properties?.maxNumberOfPinCodeCredentials),
+        SupportedPinCodeNameLengthRange(
+          PinCodeNameLengthRange(
+            min: trait.properties?.supportedPinCodeCredentialNameRange.min,
+            max: trait.properties?.supportedPinCodeCredentialNameRange.max,
+          ),
+        ),
+        SupportedPinCodeRange(
+          PinCodeLengthRange(
+            min: trait.properties?.supportedPinCodeRange.min,
+            max: trait.properties?.supportedPinCodeRange.max,
+          ),
+        ),
+      });
+    } on NoSuchMethodError catch (e, stack) {
+      Error.throwWithStackTrace(
+          ArgumentError.value(trait, 'PinCodeTrait', 'Invalid PinCodeTrait'),
+          stack);
+    }
+  }
+
+  /// Send mutation `actionPinCodeCredentialAddPinCodeCredential`
+  static Future<void> sendAddPinCodeAction(
+    Request request,
+    String id,
+    String pinCode,
+    String pinCodeName,
+  ) async {
+    final req = GmakeAddPinCodeRequest((b) {
+      b..vars.deviceId = id;
+      b..vars.pinCode = pinCode;
+      b..vars.pinCodeName = pinCodeName;
+    });
+    Repository().mutate(request, req.operation, req.vars.toJson());
+  }
+}
+
+///
+/// State-related classes representing Pin Code credentials for a PinCode Trait
+///
+class PinCodeCredentials extends State<List<PinCodeCredential>> {
+  PinCodeCredentials(rawValues)
+      : super((PinCodeCredentials).toString(), _toMappedList(rawValues)) {}
+
+  PinCodeCredential operator [](int index) => value[index];
```

Comment:
What would it take to get test coverage here?

Reply:
I mentioned in my comment here that CodeCov is not reading this line as covered when it should:
https://github.com/Yonomi/yonomi-sdk-dart/pull/81#issuecomment-1231852634